### ISSUE TITLE
Aktuell scheinen zu viele Anfragen geblockt zu werden

### ISSUE THREAD

**Issue by claell** [rexlManu/goflink-discount-checker#2]
Bei meinem Python Skript kommen heute 429 Antworten zurück. Es sieht so aus, als wenn geblockt wird, wenn zu viele Anfragen in kurzer Zeit geschickt werden, kann mich aber auch irren.

**Comment by rexlManu** [comment]
Ja die haben ein Rate Limit eingebaut bzw. Das von cloudflare aktiviert.
Man müsste das mit proxies umgehen.

Claudius Ellsel ***@***.***> schrieb am Mo., 15. Nov. 2021,
19:06:

**Comment by claell** [comment]
Ja, wobei das recht aufwändig wieder wird. Weißt du, ob man da mit genügend Wartezeit zwischen Anfragen durch kommt? Aktuell kann ich wieder anfragen schicken und teste gerade mal mit einer Wartezeit von einem Zufallswert zwischen 10 und 23 Sekunden. Scheint aktuell zumindest mehr Anfragen zuzulassen, als mit einer Zeit zwischen 1 und 2 Sekunden (da bin ich glaube ich nach vier Anfragen geblockt worden).

Sehe gerade, dass das Repository hier auf Englisch ist und wir auch im anderen Issue auf Englisch geschrieben haben. Können wir hier auch wieder machen, wenn du willst.

**Comment by claell** [comment]
Nach etwa 50 Anfragen kam mit dem aktuellen Timeout die Blockierung, zumindest ist da das Skript in einen Fehler reingelaufen, welcher genau das war, kann ich aktuell nicht sehen.

**Comment by claell** [comment]
Vermutlich bringt es nichts, ein zufälliges Zeitintervall zu verwenden, sondern es reicht aus, beim Warten über einer bestimmten Zeit zu bleiben. Aktuell teste ich mit 16 Sekunden.

**Comment by claell** [comment]
Das hat gerade geklappt, ist erfolgreich durchgelaufen. Muss man mal testen, wie sehr man das noch verkürzen kann.

**Comment by rexlManu** [comment]
Wenn du Anfragen sendest und ins RateLimit kommst, bekommst du besondere Header zurück?

**Comment by claell** [comment]
Man bekommt einen anderen Status Code. Der ist dann nicht 400, sondern 429 (was dann auch direkt der Hinweis ist, dass man gerade geblockt ist).

**Comment by claell** [comment]
Mit 14 Sekunden Wartezeit zwischen Anfragen bin ich nach 9 Malen geblockt worden. Es kommt dann auch ein Header zurück, der die Zeit angibt, wann man wieder Anfragen senden kann (sieht mir nach etwa einer Stunde aus). Ob schon vorher eine Art Warnung kommt, weiß ich nicht. Habe ich nicht getestet.

**Comment by claell** [comment]
Mit 15 Sekunden klappt es, ich schätze das ist der niedrigste Wert, wo es noch funktioniert.
